orchagent/aclorch.cpp

```diff
@@ -2447,10 +2467,124 @@ void AclOrch::doTask(Consumer &consumer)
     }
 }
 
-bool AclOrch::addAclTable(AclTable &newTable, string table_id)
+void AclOrch::getAddDeletePorts(AclTable    &newT,
+		                AclTable    &curT,
+				set<string> &addSet,
+				set<string> &delSet)
+{
+    set<string> newPortSet, curPortSet;
+
+    // Collect new ports
+    for (auto p : newT.pendingPortSet)
+    {
+	newPortSet.insert(p);
+    }
+    for (auto p : newT.portSet)
+    {
+	newPortSet.insert(p);
+    }
+
+    // Collect current ports
+    for (auto p : curT.pendingPortSet)
+    {
+	curPortSet.insert(p);
+    }
+    for (auto p : curT.portSet)
+    {
+	curPortSet.insert(p);
+    }
+
+    // Get the difference newPortSet-curPortSet and curPortSet-newPortsSet
+    std::set_difference(newPortSet.begin(), newPortSet.end(),
+		        curPortSet.begin(), curPortSet.end(),
+			std::inserter(addSet, addSet.end()));
+    std::set_difference(curPortSet.begin(), curPortSet.end(),
+		        newPortSet.begin(), newPortSet.end(),
+			std::inserter(delSet, delSet.end()));
+
+}
+
+bool AclOrch::updateAclTablePorts(AclTable &newTable, AclTable &curTable)
+{
+    sai_object_id_t    port_oid = SAI_NULL_OBJECT_ID;
+    set<string>        addPortSet, deletePortSet;
+
+    SWSS_LOG_ENTER();
+    getAddDeletePorts(newTable, curTable, addPortSet, deletePortSet);
+
+    // Lets first unbind and unlink ports to be removed
+    for (auto p : deletePortSet)
+    {
+        SWSS_LOG_NOTICE("Deleting port %s from ACL list %s",
+			p.c_str(), curTable.id.c_str());
+        if (curTable.pendingPortSet.find(p) != curTable.pendingPortSet.end())
+	{
+            SWSS_LOG_NOTICE("Removed:%s from pendingPortSet", p.c_str());
+	    curTable.pendingPortSet.erase(p);
+	} else if (curTable.portSet.find(p) != curTable.portSet.end())
+	{
+            gPortsOrch->getAclBindPortId(p, port_oid);
+	    assert(port_oid != SAI_NULL_OBJECT_ID);
+	    assert(curTable.ports.find(port_oid) != curTable.ports.end());
+	    if (curTable.ports[port_oid] != SAI_NULL_OBJECT_ID)
+	    {
+                // Unbind and unlink
+                SWSS_LOG_NOTICE("Unbind and Unlink:%s", p.c_str());
+		curTable.unbind(port_oid);
+		curTable.unlink(port_oid);
+	    }
+            SWSS_LOG_NOTICE("Removed:%s from portSet", p.c_str());
+	    curTable.portSet.erase(p);
+	}
+    }
+
+    // Now link and bind ports to be added
+    for (auto p : addPortSet)
+    {
+        SWSS_LOG_NOTICE("Adding port %s to ACL list %s",
+			p.c_str(), curTable.id.c_str());
+        Port port;
+	if (!gPortsOrch->getPort(p, port))
+	{
+	    curTable.pendingPortSet.emplace(p);
+	    continue;
+	}
+        if (!gPortsOrch->getAclBindPortId(p, port_oid))
```

Comment:
As explained, this can happen when the port object is created in portsOrch, but port is not created using SAI API. Hence we won't have OID. But as I showed you in the processAclTablePorts function, we assume that if the port object is present in portsOrch, OID will also be there. Hence, I will at both places!

Reply:
As discussed again, we instead of asserting, we are going to return false here. That is because, if the port is part of LAG, getACLBindPortId() return false.